### PR TITLE
Display all columns in a experiment

### PR DESCRIPTION
**Pull request by jdlesage**
## What changes are proposed in this pull request?

This ticket is about to fix https://github.com/mlflow/mlflow/issues/2475

Add a new service that list all columns of a experiment
In javascript, call this service when a new search is triggered.

For the future: Tags, params and metrics list are not injected in the ExperimentView component. It could be reuse to implement autocompletion in the search bar.

## How is this patch tested?

Tested on our data + unit test

all_columns

### Is this a user-facing change?
- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

All columns (parameters, tags, or metrics) are displayed in the experiment view even if the first runs does not contain values for these columns.

### What component(s) does this PR affect?

- [X] UI
- [ ] CLI
- [ ] API
- [X] REST-API
- [ ] Examples
- [X] Docs
- [X] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [X] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/server/js/src/components/ExperimentPage.js [hunk 1]
```diff
@@ -175,6 +193,20 @@ export class ExperimentPage extends Component {
...
+        this.setState({
+          metricsList: response.value.metrics,
+          tagsList: response.value.tags,
+          paramsList: response.value.params,
+        });
+      }
+      )
+      .catch((e) => {
+        Utils.logErrorAndNotifyUser(e);
+      });
```
Comment (jcuquemelle) [inline mlflow/server/js/src/components/ExperimentPage.js]: not sure about JS good practises, this code is duplicated, maybe make it a method ?

File: mlflow/protos/service.proto [hunk 2]
```diff
@@ -346,6 +346,24 @@ service MlflowService {
...
+    option (rpc) = {
+      endpoints: [{
+        method: "GET",
+        path: "/mlflow/columns/list"
+        since { major: 2, minor: 0 },
+      }, {
+        method: "GET",
+        path: "/preview/mlflow/columns/list"
+        since { major: 2, minor: 0 },
+      }],
```
[the remark this answers is not in the document]
Comment (jdlesage) [reply mlflow/protos/service.proto]: Agree, but all services define the /preview... I followed the cameleon rule

File: mlflow/server/js/src/components/ExperimentPage.js (hunk 1 above)
Comment (jdlesage) [reply mlflow/server/js/src/components/ExperimentPage.js]: Where is duplicated ? The contents of the blocks differ actually. (actually I followed the same pattern as searchRunsApi

Comment (jcuquemelle) [reply mlflow/server/js/src/components/ExperimentPage.js]: same block as lines 69 -> 81, with experimentId <- this.props.experimentId and viewType <- lifecycleFilterToRunViewType(lifecycleFilterInput)

But as I said, I don't know JS enough to know if it's doable to write a function that takes 2 arguments and implement a single block for the 2 cases

Will accept the review as is, we can discuss later when I know more JS

**Pull request merged**